.github/workflows/deploy.yaml

```diff
@@ -1,17 +1,43 @@
+# Workflow that builds a Docker image containing the model code,
+# pushes it to the GitHub Container Registry, and then optionally uses
+# that container image to run the model using an AWS Batch job.
+#
+# Images are built on every commit to a PR or main branch in order to ensure
+# that the build continues to work properly, but Batch jobs are gated behind
+# a `deploy` environment that requires manual approval from a codeowner.
+
 name: deploy
 
 on:
```

Comment:
**suggestion:** We may want to _add_ a workflow dispatch option, since often we'll need to do runs when something about the underlying data changes. Though maybe this is handled by ingesting directly from the DVC cache, since updating it would require a PR or push.

Reply:
Sounds right! I expect we'll also want to trigger runs from the main branch, too.

Reply:
Done in 2a85c146466071bfde2d319b2674c074ca2863ff.